Updated Mongodb to version 2.1

CaselIT opened the pull request:
Updated the type definitions to version 2.1 with Promise support.
Added the file `mongodb-1.4.9.d.ts` that contains the old definitions.

Also thanks to Think7 for the partial type definitions that I used as base

vvakame commented on acl/acl.d.ts:
```diff
@@ -3,7 +3,7 @@
 // Definitions: https://github.com/borisyankov/DefinitelyTyped
 
-/// <reference path="../bluebird/bluebird.d.ts" />
+/// <reference path='../es6-promise/es6-promise.d.ts' />
```
https://github.com/OptimalBits/node_acl/blob/master/package.json#L19
acl uses bluebird.

CaselIT replied:
You are right. I changed it because Travis gave an error about a double definition of Promise, since now mongodb uses es6-promise.
Do you know how can I avoid the error when reverting to bluebird in acl?
Thanks

CaselIT commented:
@vvakame I reverted the wrong promise library change in acl. To avoid the double definition of promise error I changed the reference to the old mongodb typings in acl instead.

vvakame commented:
We should wait for https://github.com/DefinitelyTyped/DefinitelyTyped/pull/7305 is merged.

CaselIT commented:
Any update?

vvakame commented:
you can use bluebird now.

CaselIT commented:
Even if the library actually uses es6-promise?https://github.com/mongodb/node-mongodb-native/blob/2.1/package.json

CaselIT commented:
@vvakame Any suggestion regarding my previous comment? Thanks

falsandtru commented:
@horiuchi could you?

horiuchi commented:
LGTM :+1: 
@vvakame you?

vvakame commented:
:+1: 
@CaselIT please resolve conflict.

CaselIT commented:
@vvakame I can't seem to resolve them correctly. I'll recreate the pull request starting from a new fork of the repository.

CaselIT commented:
@vvakame the new pull request is here https://github.com/DefinitelyTyped/DefinitelyTyped/pull/7565
